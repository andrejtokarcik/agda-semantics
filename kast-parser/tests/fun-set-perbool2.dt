data Bool : Set where
  True : Bool,
  False : Bool
;
data T : Set where
  E : T

; s True = Bool
; s False = T
; f2 = Lambda x Bool (Lambda y (s x) y)
; h = f2 False E
; g = f2 True True
