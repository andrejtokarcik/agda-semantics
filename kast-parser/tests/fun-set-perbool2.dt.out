<dbg>
    ListItem((Pi y  s x   s x ))
    ListItem((Pi y fun ( ListItem(True -> Bool) ListItem(False -> T) ) 
      #symVariable(13) fun ( ListItem(True -> Bool) ListItem(False -> T) ) 
      #symVariable(13)))
    ListItem((Pi y fun ( ListItem(True -> Bool) ListItem(False -> T) ) 
      #symVariable(20) fun ( ListItem(True -> Bool) ListItem(False -> T) ) 
      #symVariable(20)))
</dbg>
<T>
    <k>
        .K
    </k>
    <mgu>
        subst(.KList)
    </mgu>
    <data>
        Bool |-> Set 0
        E |-> T
        False |-> Bool
        T |-> Set 0
        True |-> Bool
    </data>
    <meta>
        .Map
    </meta>
    <fun>
        f2 |-> ((Lambda x Bool (Lambda y  s x  y)) @ s |-> (fun ( 
          ListItem(True -> Bool) ListItem(False -> T) ) @ .Map))
        g |->   f2 True  True 
        h |->   f2 False  E 
        s |-> (fun ( ListItem(True -> Bool) ListItem(False -> T) ) @ .Map)
          
    </fun>
    <tenv>
        f2 |-> (Pi x Bool (Pi y fun ( ListItem(True -> Bool) ListItem(
          False -> T) ) x fun ( ListItem(True -> Bool) ListItem(False -> T) ) 
          x))
        g |-> Bool
        h |-> T
        s |-> (Pi #symVariable(5) Bool Set 0)
    </tenv>
</T>
